inst/examples/shiny_interactions.R

```diff
@@ -0,0 +1,247 @@
+    if(is.null(input$addfeature)==FALSE && input$addfeature != "") {
+      geojson <- geojson$features[[seq_along(geojson$features)[sapply(geojson$features,
+              FUN = function(x) x[["id"]] == input$addfeature)]]]
+      leafletProxy("mymap") %>%
+      addFeatureGeoJSON(geojson, layerId ='geojsonlayer') # can use a list (slow)
+      if (length(addedData$df$ids) > 1) {
+        removedData$df <- removedData$df[-c(which(removedData$df$ids==input$addfeature)),]
+      }
+      addedData$df <- rbind(addedData$df,allCountries[which(allCountries$ids==input$addfeature),])
+      updateSelectizeInput(session, 'setstyle', choices = addedData$df$ids, server = TRUE, selected=NULL)
+      updateSelectizeInput(session, 'removefeature', choices = addedData$df$ids, server = TRUE, selected=NULL)
+      updateSelectizeInput(session, 'addfeature', choices = removedData$df$ids, server = TRUE, selected=NULL)
+    }
+  })
+  observeEvent(input$clearGeojson, {
+    leafletProxy("mymap") %>% removeGeoJSON(layerId ='geojsonlayer')
+  })
+  observeEvent(input$addPopup, {
+    content <- paste(sep = "<br/>",
+                     "<b><a href='http://www.samurainoodle.com'>Samurai Noodle</a></b>",
+                     "606 5th Ave. S",
+                     "Seattle, WA 98138"
+    )
+    leafletProxy("mymap") %>%  addPopups(-122.327298, 47.597131, content,
+                         options = popupOptions(closeButton = TRUE))
+  })
+  observeEvent(input$clearPopup, {
+    leafletProxy("mymap") %>% clearPopups()
+  })
+  observeEvent(input$addbasemap, {
+    leafletProxy("mymap") %>% addProviderTiles("Acetate.terrain",options = providerTileOptions(noWrap = TRUE,zIndex=0),layerId="basemap")
+  })
+  observeEvent(input$clearbasemap, {
+    leafletProxy("mymap") %>% removeTiles("basemap") %>% removeTiles("wms_protectedArea")
+  })
+  observeEvent(input$clear, {
+    leafletProxy("mymap") %>% clearTiles()
+  })
+  observeEvent(input$mymap_geojson_mouseover, {
+        leafletProxy("mymap") %>%
+          styleFeatureGeoJSON(layerId ='geojsonlayer', featureId = input$mymap_geojson_mouseover$featureId,
+                              style = list(weight=5,color="black")) # can use a list
+  })
+  observeEvent(input$mymap_geojson_mouseout, {
+        leafletProxy("mymap") %>%
+          styleFeatureGeoJSON(layerId ='geojsonlayer', featureId = input$mymap_geojson_mouseout$featureId,
+                              style = '{"weight": 1, "color": "#555555"}') # or string
+  })
+  observeEvent(input$addBermuda, {
+    # geoJSON layer must already be added
+    # simplified example, not added to addedData
+    leafletProxy("mymap") %>%
+      addFeatureGeoJSON(as.character(minify(bermudaTriangle)), layerId ='geojsonlayer') # can use a GeoJSON string (as.character)
+  })
+  observeEvent(input$removeBermuda, {
+    leafletProxy("mymap") %>%
+      removeFeatureGeoJSON(layerId ='geojsonlayer', featureId = "Bermuda Triangle")
+  })
+}
+shinyApp(ui, server)
+
```

Comment:
The example is very nice :+1: The interaction (mouseover) is a bit slow, though. I didn't investigate this issue.